src/Common/Scheduler/Nodes/IOResourceManager.cpp

```diff
@@ -217,8 +217,7 @@ IOResourceManager::Workload::~Workload()
     }
     catch (...)
     {
-        throw Exception(ErrorCodes::LOGICAL_ERROR, "Unexpected error in IOResourceManager: {}",
-            getCurrentExceptionMessage(/* with_stacktrace = */ true));
+        tryLogCurrentException("Workload");
```

Comment:
maybe `chassert` to cause a clear failure in tests?

Reply:
Good idea, although it's not happening in tests yet